src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkService.java

```diff
@@ -45,6 +45,14 @@ public interface SnowflakeSinkService {
    */
   void insert(final SinkRecord record);
 
+  /**
+   * Insert a JSON record into the buffer
+   *
+   * @param record record content
+   * @param isFirstRowInBatch indicates whether the given record is the first record in a batch
```

Comment:
its good to indicate if its first record in the batch that kafka sends or first record in insert rows API batch.